src/main/Main.hs

```diff
@@ -385,7 +385,7 @@ commandLineHandler currentDir progName isInterpreter = complicatedOptions
                     cleanCmd
                     cleanOptsParser
         addCommand' "list-dependencies"
-                    "List the dependencies"
+                    "DEPRECATED: Use ls dependencies instead. Will be removed in next major version."
```

Comment:
While it's a safe bet, no other deprecation notice specifies a time :/

Reply:
We actually haven't been so good at removing deprecated stuff.

Reply:
I'd say we should have `list-dependencies` output a warning (to stderr) that it may be removed at some point.  Thing is, people do use it in scripts.  We should probably wait until stack 2.0 or 3.0 to remove it.  Could hide it from the --help output next version, though.